docs/source/json-api/index.rst

```diff
@@ -321,33 +323,44 @@ and archives the one above, the same stream will eventually produce::
             },
             "signatories": ["Alice"],
             "contractId": "#2:2",
-            "templateId": "f95486336ffb3c982319625bed0c88f68799b780b26b558b1e119277614ed634:Iou:Iou"
-        }],
-        "archived": ["#1:0"]
-    }
+            "templateId": "6cc82609ede6576e5092e8c89a2c7a658efe15ae1347fc38eb316f787fa132bc:Iou:Iou"
+        }
+    }, {
+        "archived": "#1:0"
+    }]
+
+Aside from ``"created"`` and ``"archived"`` elements, ``"error"``
+elements may appear, which contain a string describing the error.  The
+stream will continue in these cases, rather than terminating.
 
 Some notes on behavior:
 
-1. Each result object means "this is what would have changed if you just
+1. Each result array means "this is what would have changed if you just
    polled ``/contracts/search`` iteratively."  In particular, just as
    polling search can "miss" contracts (as a create and archive can be
    paired between polls), such contracts may or may not appear in any
    result object.
 
 2. No ``archived`` ever contains a contract ID occurring within an
-   ``created`` in the same object.  So, for example, supposing you are
+   ``created`` in the same array.  So, for example, supposing you are
    keeping an internal map of active contracts, you can apply the
    ``created`` first or the ``archived`` first and be guaranteed to get
    the same results.
 
-3. You will almost certainly receive contract IDs in the ``archived``
-   set that you never received an ``created`` for.  These are contracts
-   that query filtered out, but for which the server no longer is aware
-   of that.  You can safely ignore these.  However, such "phantom
-   archives" *are* guaranteed to represent an actual archival *on the
-   ledger*, so if you are keeping a more global dataset outside the
-   context of this specific search, you can use that archival
-   information as you wish.
+3. You will almost certainly receive contract IDs in ``archived`` that
```

Comment:
Right, archived events don't have the payload (template args) to apply the search filter.
Should we explain this in the doc, so we don't get asked why we don't filter these archived events?

Reply:
I've already explained this in a more abstract, interface-y sense; I hope anyone who cared enough about the implementation details beyond that would dig and find your comment.